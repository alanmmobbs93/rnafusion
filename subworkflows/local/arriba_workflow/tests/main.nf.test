nextflow_workflow {

    name "Test Subworkflow ARRIBA_WORKFLOW"
    script "../main.nf"
    //config "./nextflow.config"
    workflow "ARRIBA_WORKFLOW"
    tag "subworkflow"
    tag "arriba"
    tag "arriba/arriba"
    tag "samtools"
    tag "samtools/index"
    tag "samtools/sort"
    tag "samtools/view"
    tag "star"
    tag "star/align"


    test("ARRIBA_WORKFLOW - Homo sapiens chr22") {

        when {
            setup {
                // Create genome index for STAR
                run("STAR_GENOMEGENERATE") {
                    script "../../../../modules/nf-core/star/genomegenerate/main"
                    process {
                        """
                        // FASTA
                        input[0] = Channel.fromPath(
                                "https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/ensembl/Homo_sapiens.GRCh38.102.chr4.1700000-54900000.fa", checkIfExist: true
                            )
                            .map{ [[id: it.name], it ]}


                        // GTF
                        input[1] = Channel.fromPath(
                                "https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/ensembl/Homo_sapiens.GRCh38.102.chr4.gtf", checkIfExists: true
                            )
                            .map{ [[id: it.name], it ]}
                        """
                    }
                }
            }

            workflow {
                """
                // ch_reads
                input[0] = Channel.of(
                    [
                        [ id: "test_fastqs" ],
                        [
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/human/reads_1.fq.gz", checkIfExists: true),
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/human/reads_2.fq.gz", checkIfExists: true)
                        ]
                    ] )

                // ch_gtf
                input[1] =
                    Channel.fromPath(
                        "https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/ensembl/Homo_sapiens.GRCh38.102.chr4.gtf", checkIfExists: true
                    )
                    .map{ [ [ id: it.name ], it ] }

                // ch_fasta
                input[2] =
                    Channel.fromPath(
                        "https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/ensembl/Homo_sapiens.GRCh38.102.chr4.1700000-54900000.fa", checkIfExist: true
                    )
                    .map{ [ [ id: it.name ], it ] }

                // ch_starindex_ref
                input[3] = STAR_GENOMEGENERATE.out.index.map{ [ [id: "star_index"], it ] }

                // ch_arriba_ref_blacklist
                input[4] =
                    Channel.fromPath(
                        "https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/arriba/blacklist_hg38_GRCh38_v2.4.0.tsv.gz", checkIfExist: true
                    )
                    .map{ [ [ id: it.name ], it ] }

                // ch_arriba_ref_known_fusions
                input[5] =
                    Channel.fromPath(
                        "https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/arriba/known_fusions_hg38_GRCh38_v2.4.0.tsv.gz", checkIfExist: true
                    )
                    .map{ [ [ id: it.name ], it ] }

                // ch_arriba_ref_cytobands
                input[6] =
                    Channel.fromPath(
                        "https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/arriba/cytobands_hg38_GRCh38_v2.4.0.tsv", checkIfExist: true
                    )
                    .map{ [ [ id: it.name ], it ] }

                // ch_arriba_ref_protein_domains
                input[7] =
                    Channel.fromPath(
                        "https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/arriba/cytobands_hg38_GRCh38_v2.4.0.tsv", checkIfExist: true
                    )
                    .map{ [ [ id: it.name ], it ] }

                // arriba (boolean)
                input[8] = true

                // all (boolean)
                input[9] = true

                // fusioninspector_only  (boolean)
                input[10] = false

                // star_ignore_sjdbgtf (boolean)
                input[11] = false

                // seq_center (string)
                input[12] = 'test_center'

                // arriba_fusions (path)
                input[13] = null

                // cram (array??)
                input[14] = [ 'arriba' ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success }/*,
                { assert snapshot(file( workflow.out.versions[0] )).match('versions' )                                                },
                { assert snapshot(file( workflow.out.rnaseq_metrics[0][1]     ).readLines()[4..-1]).md5().match('rnaseq_metrics'    ) },
                { assert snapshot(file( workflow.out.duplicate_metrics[0][1]  ).readLines()[4..-1]).md5().match('duplicate_metrics' ) },
                { assert snapshot(file( workflow.out.insertsize_metrics[0][1] ).readLines()[4..-1]).md5().match('insertsize_metrics') }*/
            )
        }
    }

}
