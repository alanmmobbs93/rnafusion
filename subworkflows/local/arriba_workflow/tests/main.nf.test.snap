{
    "ARRIBA_WORKFLOW - Homo sapiens - FASTQs chr4": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_fastqs"
                        },
                        "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/da/0878f3451a0f8db34ffd9b178974e6/test_fastqs.arriba.fusions.tsv"
                    ]
                ],
                "1": [
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/da/0878f3451a0f8db34ffd9b178974e6/test_fastqs.arriba.fusions.discarded.tsv"
                ],
                "2": [
                    [
                        {
                            "id": "test_fastqs"
                        },
                        "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/c8/482130c0382c0d0318b9e98d0650da/test_fastqs_star_for_arriba_sorted.cram",
                        "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/b5/57dffeefbc4801b0f7c4af2c8530e6/test_fastqs_star_for_arriba_sorted.cram.crai"
                    ]
                ],
                "3": [
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/da/0878f3451a0f8db34ffd9b178974e6/versions.yml",
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/5b/18880ae27c2497bddad57c4b7ba766/versions.yml",
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/b5/57dffeefbc4801b0f7c4af2c8530e6/versions.yml",
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/c8/482130c0382c0d0318b9e98d0650da/versions.yml",
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/48/58af6204cc601a1a0e6e826a117f31/versions.yml"
                ],
                "cram_index": [
                    [
                        {
                            "id": "test_fastqs"
                        },
                        "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/c8/482130c0382c0d0318b9e98d0650da/test_fastqs_star_for_arriba_sorted.cram",
                        "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/b5/57dffeefbc4801b0f7c4af2c8530e6/test_fastqs_star_for_arriba_sorted.cram.crai"
                    ]
                ],
                "fusions": [
                    [
                        {
                            "id": "test_fastqs"
                        },
                        "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/da/0878f3451a0f8db34ffd9b178974e6/test_fastqs.arriba.fusions.tsv"
                    ]
                ],
                "fusions_fail": [
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/da/0878f3451a0f8db34ffd9b178974e6/test_fastqs.arriba.fusions.discarded.tsv"
                ],
                "versions": [
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/da/0878f3451a0f8db34ffd9b178974e6/versions.yml",
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/5b/18880ae27c2497bddad57c4b7ba766/versions.yml",
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/b5/57dffeefbc4801b0f7c4af2c8530e6/versions.yml",
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/c8/482130c0382c0d0318b9e98d0650da/versions.yml",
                    "/workspace/rnafusion/.nf-test/tests/7899df1159e4d52047799028a4a24a9a/work/48/58af6204cc601a1a0e6e826a117f31/versions.yml"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-09T18:02:58.142697295"
    },
    "ARRIBA_WORKFLOW - Homo sapiens - FASTQs chr4 - cram = []": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_fastqs"
                        },
                        "test_fastqs.arriba.fusions.tsv:md5,8f39789c4428e81eb9a8d0e54c34c43d"
                    ]
                ],
                "1": [
                    "test_fastqs.arriba.fusions.discarded.tsv:md5,b804c1ed5b01d34163f5c0b2f6810f98"
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,24030f38976402fad0861e6ec99ee6b6",
                    "versions.yml:md5,439bbb92ff0a83f1e278fc396e9d8ce9"
                ],
                "cram_index": [
                    
                ],
                "fusions": [
                    [
                        {
                            "id": "test_fastqs"
                        },
                        "test_fastqs.arriba.fusions.tsv:md5,8f39789c4428e81eb9a8d0e54c34c43d"
                    ]
                ],
                "fusions_fail": [
                    "test_fastqs.arriba.fusions.discarded.tsv:md5,b804c1ed5b01d34163f5c0b2f6810f98"
                ],
                "versions": [
                    "versions.yml:md5,24030f38976402fad0861e6ec99ee6b6",
                    "versions.yml:md5,439bbb92ff0a83f1e278fc396e9d8ce9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-09T18:15:28.965848873"
    }
}